airflow/api_fastapi/core_api/routes/public/import_error.py

```diff
@@ -36,18 +42,38 @@
     ImportErrorResponse,
 )
 from airflow.api_fastapi.core_api.openapi.exceptions import create_openapi_http_exception_doc
+from airflow.api_fastapi.core_api.security import (
+    AccessView,
+    GetUserDep,
+    requires_access_dag,
+    requires_access_view,
+)
+from airflow.models import DagModel
 from airflow.models.errors import ParseImportError
 
+if TYPE_CHECKING:
+    from sqlalchemy.orm import Session
+
+REDACTED_STACKTRACE = "REDACTED - you do not have read permission on all DAGs in the file"
 import_error_router = AirflowRouter(tags=["Import Error"], prefix="/importErrors")
 
 
+def get_file_dag_ids(session: Session, filename: str) -> set[str]:
+    return {dag_id for dag_id in session.scalars(select(DagModel.dag_id).where(DagModel.fileloc == filename))}
+
+
 @import_error_router.get(
     "/{import_error_id}",
-    responses=create_openapi_http_exception_doc([status.HTTP_404_NOT_FOUND]),
+    responses=create_openapi_http_exception_doc([status.HTTP_403_FORBIDDEN, status.HTTP_404_NOT_FOUND]),
```

Comment:
```suggestion
    responses=create_openapi_http_exception_doc([status.HTTP_404_NOT_FOUND]),
```

Forbidden is already part of the base router.